### ISSUE TITLE
Feedbacks about a first time experience

### ISSUE THREAD

**Issue by Nutriz** [godotengine/godot-proposals#742]
I will try to fill in the template but this issue is a bit special, in sense of I'm reporting a guy (not me) who complains about his first experience in Godot. I was happy to see this [nice project](https://ldjam.com/events/ludum-dare/46/otherworldlyflavors) was made with Godot, I had a very fun and productive moment to use Godot for this last ludum dare, so my curiosity ask him how was his first time with Godot, and...and...

> "I did not have a good time with Godot. There were so many gotchas and random hiccups with bad google/documentation resolutions"

:sob: :sob: :sob: 

This sentence was followed by a list of points (see below). I don't have really proposal for there points but I think is interesting to have first time experience feedbacks.

**Describe the project you are working on:**
LD46, 2D card game

**Describe the problem or limitation you are having in your project:**
* I spent 30 minutes trying to figure out how to center text in a rich text label.
* I spent an hour trying to get the icon for the exe changed and still couldn’t do it.
* The engine text editor was a complete pain to work with. It didn’t open new files automatically when created. The tabs on the top and panel to the left for scripts vs scene was frustrating as scenes and scripts often had similar names.
* F2 to change to 3d view instead of renaming an object drove me mad.
* Changing a nodes placement in the tree breaks all your scripts and there is no tool that automatically looks in all scripts for if that object has been accessed and tries to auto fix the path for it.
* Sure it(Godot) is a tiny download, but that’s because they don’t download the packaging templates with the base engine. So when you actually want to export then you get another half gig of data. It just didn’t feel ready for prime time still.

**Comment by KoBeWi** [comment]
> I spent 30 minutes trying to figure out how to center text in a rich text label.
> I spent an hour trying to get the icon for the exe changed and still couldn’t do it.

Googling it for the answer takes few minutes. Also it's in the docs.

> F2 to change to 3d view instead of renaming an object drove me mad.

The shortcut for both renaming and 3D view can be changed, so...

> Changing a nodes placement in the tree breaks all your scripts and there is no tool that automatically looks in all scripts for if that object has been accessed and tries to auto fix the path for it.

There are NodePaths for that.

> Sure it(Godot) is a tiny download, but that’s because they don’t download the packaging templates with the base engine.

Nothing can be done about it. You could ship your game files along with Godot executable as a workaround.

The script editor complaint is somewhat relevant, but the others come from the inability to search for information (?). Like, you can find all necessary stuff in the docs or on google and rebind the shortcuts if they trigger you. Not much can be done on Godot side about all this, so I don't think this feedback is really useful...

**Comment by Xrayez** [comment]
> > Changing a nodes placement in the tree breaks all your scripts and there is no tool that automatically looks in all scripts for if that object has been accessed and tries to auto fix the path for it.
> 
> There are NodePaths for that.

@KoBeWi I don't think I follow, do you suggest that `NodePath` references get automatically refactored within the scripts themselves? I can't seem to find a way to achieve that, in those cases it requires manual refactor. And sure you have `onready` vars for that, I'm not talking about this kind of refactor. Perhaps there's a hidden feature?

I've tried `get_node(@"path/to/node")`, `$path/to/node` and any other combination, no success.

It also does seem to work when you `export` the node path itself.

**Comment by KoBeWi** [comment]
@Xrayez Yeah, I mean you can export NodePath and it gets automatically updated when node is moved/renamed etc. It's the only way. We could make it more convenient by allowing to export Node directly (which would internally be stored as NodePath); using `get_node()` each time on exported path is rather annoying.

Also auto-refactoring tool for changed paths is difficult to do. We'd either need to remember previous path for each node (or rather whole path history, there's no telling when you need to update script) or scan scripts for using node paths and auto-update them (which is a problem, because it would only work with tightly tied scripts and static paths; also no tool for auto-editing scripts exists yet).

**Comment by Jummit** [comment]
https://docs.godotengine.org/de/latest/getting_started/workflow/export/changing_application_icon_for_windows.html

**Comment by Calinou** [comment]
Unfortunately, we can't simplify the icon changing process until someone makes an easy-to-integrate C/C++ library to edit PE executable metadata (including the icon).
